coordinators/simple/coordinator.go

```diff
@@ -25,11 +28,17 @@ type Simple struct {
 
 	flows    map[iface.FlowID]*FlowDetails
 	mu_flows sync.RWMutex // to make the map thread-safe
+
+	numParallelIntegrityCheckTasks int
 }
 
-func NewSimpleCoordinator() *Simple {
+func NewSimpleCoordinator(numParallelIntegrityCheckTasks int) *Simple {
```

Comment:
this feels like it should be a CoordinatorSettings struct rather than an int param

Reply:
we don't have one right now, so didn't think it needed to be there yet. i think this is an easy refactor later